Src/Base/AMReX_Utility.cpp

```diff
@@ -145,6 +145,34 @@ amrex::Concatenate (const std::string& root,
     return result.str();
 }
 
+bool
+amrex::is_next_non_space(std::istream& is, char c)
```

Comment:
I am sure this works.  But C++ provides a simpler solution,  `std::ws1.  https://en.cppreference.com/w/cpp/io/manip/ws

You want want too look at how we do a similar thing here.  https://github.com/AMReX-Codes/amrex/blob/development/Src/Base/AMReX_IntVect.cpp#L42